modules/features/taskerplugin/src/main/java/au/com/shiftyjelly/pocketcasts/taskerplugin/controlplayback/ActionRunnerControlPlayback.kt

```diff
@@ -44,8 +51,38 @@ class ActionRunnerControlPlayback : TaskerPluginRunnerActionNoOutput<InputContro
                 }
             }
             InputControlPlayback.PlaybackCommand.PlayNextInQueue -> playbackManager.playNextInQueue(PlaybackManager.PlaybackSource.TASKER)
+            InputControlPlayback.PlaybackCommand.SetPlaybackSpeed -> {
+                val speed = input.regular.playbackSpeed?.toDoubleOrNull() ?: return TaskerPluginResultError(ERROR_INVALID_PLAYBACK_SPEED_PROVIDED, context.getString(R.string.playback_speed_not_valid, input.regular.playbackSpeed))
+
+                val clippedToRangeSpeed = speed.coerceIn(0.5, 3.0)
+                val roundedSpeed = round(clippedToRangeSpeed * 10.0) / 10.0
+                context.updateEffects { playbackSpeed = roundedSpeed }
+            }
+            InputControlPlayback.PlaybackCommand.SetTrimSilenceMode -> {
+                val trimSilenceMode = input.regular.trimSilenceModeEnum ?: return TaskerPluginResultError(ERROR_INVALID_TRIM_SILENCE_MODE_PROVIDED, context.getString(R.string.trim_silence_mode_not_valid, input.regular.trimSilenceMode))
+
+                context.updateEffects { trimMode = trimSilenceMode }
+            }
+            InputControlPlayback.PlaybackCommand.SetVolumeBoost -> {
+                val volumeBoostEnabled = input.regular.volumeBoostEnabled?.toBooleanStrictOrNull() ?: return TaskerPluginResultError(ERROR_INVALID_VOLUME_BOOST_VALUE_PROVIDED, context.getString(R.string.volume_boost_enabled_not_valid, input.regular.volumeBoostEnabled))
+
+                context.updateEffects { isVolumeBoosted = volumeBoostEnabled }
+            }
         }
 
         return TaskerPluginResultSucess()
     }
+
+    private fun Context.updateEffects(updater: PlaybackEffects.() -> Unit) {
+
+        val playbackEffects = settings.getGlobalPlaybackEffects()
+        playbackEffects.updater()
+
+        val currentPodcast = playbackManager.getCurrentPodcast()
+        if (currentPodcast != null) {
+            podcastManager.updateEffects(currentPodcast, playbackEffects)
+        }
+
+        playbackManager.updatePlayerEffects(playbackEffects)
```

Comment:
Thank you very much for the feedback and detailed response! 

Indeed, I shouldn't be getting the effects from the global settings, that was silly of me.

Since this action only controls playback though, I want to stay clear of updating any settings. So, I really don't want to update either podcast or global effects. I just want to update the current playback effects (maybe I or someone else can add actions for settings later).

So, just so I get this right, there are 3 possible places where effects are applied, right?

1. Podcast effects
2. Global effects
3. Playback effects

Did I understand it right that using

`playbackManager.updatePlayerEffects(effects)`

should update the player UI if no effect is applied globally or by the podcast itself? It's actually not doing that for me... If I use that to update the playback effects, the new effects are correctly heard but the UI doesn't update correctly.

Also, do you think it's possible to get the current effects applied to the player (not the podcast) and update those?

Ultimately I would like to do something like:

```
private fun Context.updateEffects(updater: PlaybackEffects.() -> Unit) {
      val playbackEffects = playbackManager?.playbackEffects ?: PlaybackEffects() //this part does not exist
      playbackEffects.updater()
      playbackManager.updatePlayerEffects(playbackEffects)
}
```

Thanks again!

Reply:
Ok, you're totally right :) I was misunderstanding part of how it works but I think I got it now.

I just updated the commit now and tested it, so I think it's working as expected now.

Thanks again!